azurerm/resource_arm_storage_blob.go

```diff
@@ -556,6 +600,13 @@ func resourceArmStorageBlobRead(d *schema.ResourceData, meta interface{}) error
 
 	container := blobClient.GetContainerReference(storageContainerName)
 	blob := container.GetBlobReference(name)
+
+	err = blob.GetProperties(nil)
+	if err != nil {
+		return fmt.Errorf("Error getting properties of blob %s (container %s, storage account %s): %+v", name, storageContainerName, storageAccountName, err)
+	}
+	d.Set("content_type", blob.Properties.ContentType)
```

Comment:
can we nil-check if `Properties != nil` here?

Reply:
`Properties` is not a pointer type: https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/vendor/github.com/Azure/azure-sdk-for-go/storage/blob.go#L34. Actually both `Properties` and `ContentType` are not pointer types.